packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayer.m

```diff
@@ -421,6 +426,216 @@ - (void)setPlaybackSpeed:(double)speed error:(FlutterError *_Nullable *_Nonnull)
+  }
+
+  AVAsset *asset = currentItem.asset;
+
+  // First, try to get tracks from media selection (for HLS streams)
+  AVMediaSelectionGroup *audioGroup =
+      [asset mediaSelectionGroupForMediaCharacteristic:AVMediaCharacteristicAudible];
+  if (audioGroup && audioGroup.options.count > 0) {
+    // Cache the options array for later use in selectAudioTrack
+    _cachedAudioSelectionOptions = audioGroup.options;
+
+    NSMutableArray<FVPMediaSelectionAudioTrackData *> *mediaSelectionTracks =
+        [[NSMutableArray alloc] init];
+    AVMediaSelectionOption *currentSelection = nil;
+    if (@available(iOS 11.0, macOS 10.13, *)) {
+      AVMediaSelection *mediaSelection = currentItem.currentMediaSelection;
+      currentSelection = [mediaSelection selectedMediaOptionInMediaSelectionGroup:audioGroup];
+    } else {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+      currentSelection = [currentItem selectedMediaOptionInMediaSelectionGroup:audioGroup];
+#pragma clang diagnostic pop
+    }
+
+    for (NSInteger i = 0; i < audioGroup.options.count; i++) {
+      AVMediaSelectionOption *option = audioGroup.options[i];
+
+      // Skip nil options
+      if (!option || [option isKindOfClass:[NSNull class]]) {
+        continue;
+      }
+
+      NSString *displayName = option.displayName;
+
+      NSString *languageCode = nil;
+      if (option.locale) {
+        languageCode = option.locale.languageCode;
+      }
+
+      NSString *commonMetadataTitle = nil;
+      NSArray<AVMetadataItem *> *titleItems =
+          [AVMetadataItem metadataItemsFromArray:option.commonMetadata
+                                         withKey:AVMetadataCommonKeyTitle
+                                        keySpace:AVMetadataKeySpaceCommon];
+      if (titleItems.count > 0 && titleItems.firstObject.stringValue) {
+        commonMetadataTitle = titleItems.firstObject.stringValue;
+      }
+
+      BOOL isSelected = (currentSelection == option) || [currentSelection isEqual:option];
+
+      FVPMediaSelectionAudioTrackData *trackData =
+          [FVPMediaSelectionAudioTrackData makeWithIndex:i
+                                             displayName:displayName
+                                            languageCode:languageCode
+                                              isSelected:isSelected
+                                     commonMetadataTitle:commonMetadataTitle];
+
+      [mediaSelectionTracks addObject:trackData];
+    }
+
+    // Always return media selection tracks when there's a media selection group
+    // even if all options were nil/invalid (empty array)
+    return [FVPNativeAudioTrackData makeWithAssetTracks:nil
+                                   mediaSelectionTracks:mediaSelectionTracks];
+  }
+
+  // If no media selection group or empty, try to get tracks from AVAsset (for regular video files)
+  NSArray<AVAssetTrack *> *assetAudioTracks = [asset tracksWithMediaType:AVMediaTypeAudio];
```

Comment:
Also, the tracking loading API seems to be asynchronous and I don't see where the tracks are being loaded here.